Manito/Manito/Screens/CreateRoom/ViewModel/CreateRoomViewModel.swift

```diff
@@ -106,84 +164,91 @@ final class CreateRoomViewModel: BaseViewModelType {
             })
             .store(in: &self.cancellable)
         
-        let currentNextStep = input.nextButtonDidTap
-            .map { [weak self] step -> CurrentNextStep in
-                guard let self = self else { return (step, step.next()) }
-                return self.runActionByStep(step: step)
-            }
-            .eraseToAnyPublisher()
-        
-        let previousStep = input.backButtonDidTap
-            .map { [weak self] step -> CreateRoomStep in
-                guard let self = self else { return step }
-                return self.previous(step: step)
+        let roomInfo = Publishers.CombineLatest3(self.titleSubject, 
+                                                 self.capacitySubject,
+                                                 self.dateRangeSubject)
+            .map { title, capacity, date in
+                return RoomInfo(title: title, 
+                                capacity: capacity,
+                                dateRange: date)
             }
             .eraseToAnyPublisher()
         
-        return Output(title: self.titleSubject,
+        return Output(currentStep: currentStep,
+                      counts: mergeCount,
                       fixedTitleByMaxCount: fixedTitle,
-                      capacity: self.capacitySubject,
-                      dateRange: self.dateRangeSubject,
+                      capacity: self.capacitySubject.eraseToAnyPublisher(),
                       isEnabled: isEnabled,
-                      currentNextStep: currentNextStep, previousStep: previousStep,
-                      roomId: self.roomIdSubject)
+                      roomId: self.roomIdSubject.eraseToAnyPublisher(), 
+                      roomInfo: roomInfo)
     }
     
     // MARK: - init
     
-    init(createRoomService: CreateRoomService) {
-        self.createRoomService = createRoomService
+    init(usecase: CreateRoomUsecase) {
+        self.usecase = usecase
     }
     
     // MARK: - func
     
+    private func cutTextByMaxCount(text: String, maxCount: Int) -> String {
+        let isOverMaxCount = self.isOverMaxCount(titleCount: text.count, maxCount: maxCount)
+        
+        if isOverMaxCount {
+            let endIndex = text.index(text.startIndex, offsetBy: maxCount)
+            let fixedText = text[text.startIndex..<endIndex]
+            return String(fixedText)
+        }
+        return text
+    }
+    
     private func isOverMaxCount(titleCount: Int, maxCount: Int) -> Bool {
-        if titleCount > maxCount { return true }
-        else { return false }
+        return titleCount > maxCount ? true : false
```

Comment:
p1;

뒤의 삼항연산자를 삭제해도 될 것 같아요!
```suggestion
        return titleCount > maxCount
```

Reply:
이 부분은
titleCount 가 maxCount 보다 크면 true를 작다면 false 를 반환하는 함순데 혹시 삭제해도 되는 이유가 있을까요..??

Reply:
`titleCount > maxCount` 자체가 이미 titleCount 가 maxCount 보다 크면 true를 작다면 false 를 반환하는 형식이기 때문입니다!

Reply:
헉 그렇네요 그렇네요 제가 오히려 잘못생각하고 있었네요!!